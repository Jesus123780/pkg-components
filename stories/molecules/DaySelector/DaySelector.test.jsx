import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { DaySelector } from './index';

describe('DaySelector', () => {
  const days = [
    { day: 1, name: 'Monday' },
    { day: 2, name: 'Tuesday' },
    // ... add more days as needed
  ];

  it('renders correctly', () => {
    const { container } = render(
      <DaySelector days={days} selectedDays={[1]} handleDaySelection={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });

  it('calls handleDaySelection on circle click', () => {
    const handleDaySelection = jest.fn();
    const { getByText } = render(
      <DaySelector days={days} selectedDays={[1]} handleDaySelection={handleDaySelection} />
    );

    fireEvent.click(getByText('Monday')); // Assuming 'Monday' is in the list

    expect(handleDaySelection).toHaveBeenCalledWith(1);
  });
});
