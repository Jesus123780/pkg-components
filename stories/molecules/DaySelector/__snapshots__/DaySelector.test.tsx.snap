// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DaySelector renders correctly 1`] = `
<div>
  <div
    class="sc-bdfCDU bfuMbi"
  >
    Monday
  </div>
  <div
    class="sc-bdfCDU jFHnvP"
  >
    Tuesday
  </div>
</div>
`;
